libcontainer/container_linux.go

```diff
@@ -1124,6 +1144,17 @@ func (c *linuxContainer) runType() (Status, error) {
 		}
 		return Stopped, newSystemErrorWithCausef(err, "sending signal 0 to pid %d", pid)
 	}
+	// check if the init process is still the same process as the initial one,
+	// it could happen that the original process has exited and a new process
+	// has been created with the same pid, in this case, the container would
+	// already be stopped.
```

Comment:
I'd move this comment to before `isProcessUnique` (or whatever we want to call that function).  No need to interrupt the flow here.